### PR TITLE
Surgery code improvements

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

* Changes a lot of things about surgeries to hopefully bring it up to more modern code standards.
* Removes a ton of single-letter vars used in checking surgeries on people.
* Makes use of continue/break in for() loops.
* Properly documents the vars on surgeries
* Turns 'ignore clothes', 'self operating', 'lying required', 'require limb' and 'require real limb' from vars into surgery flags
* Removes a lot re-defines of target_mobtype being set to human, as that's the base anyways.
* Also tries to organize the vars on each surgery a bit.
* Makes the surgery initiator hopefully a little bit more sane
* Removes the surgery's can_cancel and stomach pump's accumulated_experience vars, as they were entirely unused.

## Why It's Good For The Game

I looked at surgery code and couldn't stand it, this is hopefully helping bring it to something we can stand.
This however doesn't touch the individual surgery steps.

## Changelog

im exhausted i don't know if this has in-game effects

### CODE REVIEW

File: code/datums/components/surgery_initiator.dm [hunk 1]
```diff
@@ -292,19 +289,15 @@
 		var/mob/living/carbon/carbon_target = target
 		affecting_limb = carbon_target.get_bodypart(check_zone(selected_zone))
 
-	if (surgery.requires_bodypart == isnull(affecting_limb))
```
Comment (JohnFulpWillard) [inline code/datums/components/surgery_initiator.dm:295]: if(TRUE == TRUE)

**Comment by ExcessiveUseOfCobblestone** [comment]
I'm making a soyface rn

File: code/modules/surgery/surgery.dm [hunk 2]
```diff
@@ -1,31 +1,46 @@
 /datum/surgery
+	///The name of the surgery operation
 	var/name = "surgery"
-	var/desc = "surgery description"
+	///The description of the surgery, what it does.
+	var/desc
+
+	///Selection: SURGERY_IGNORE_CLOTHES | SURGERY_SELF_OPERABLE | SURGERY_REQUIRE_RESTING | SURGERY_REQUIRE_LIMB | SURGERY_REQUIRES_REAL_LIMB
```
Comment (Mothblocks) [inline code/modules/surgery/surgery.dm:7]: Comment what the file that has these is so you can read the documentation there as well

File: code/modules/mob/living/carbon/alien/special/alien_embryo.dm [hunk 3]
```diff
@@ -68,10 +68,13 @@
...
-		for(var/datum/surgery/S in owner.surgeries)
-			if(S.location == BODY_ZONE_CHEST && istype(S.get_surgery_step(), /datum/surgery_step/manipulate_organs/internal))
-				AttemptGrow(0)
-				return
+		for(var/datum/surgery/operations as anything in owner.surgeries)
+			if(operations.location != BODY_ZONE_CHEST)
+				continue
+			if(!istype(operations.get_surgery_step(), /datum/surgery_step/manipulate_organs/internal))
+				continue
+			AttemptGrow(0)
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/alien/special/alien_embryo.dm:76]: Named parameter + probably FALSE?

File: code/modules/surgery/healing.dm [hunk 4]
```diff
@@ -63,8 +64,8 @@
 		var/datum/surgery/healing/the_surgery = surgery
 		if(!the_surgery.antispam)
 			display_results(user, target, span_notice("You attempt to patch some of [target]'s [woundtype]."),
-		span_notice("[user] attempts to patch some of [target]'s [woundtype]."),
-		span_notice("[user] attempts to patch some of [target]'s [woundtype]."))
+			span_notice("[user] attempts to patch some of [target]'s [woundtype]."),
+			span_notice("[user] attempts to patch some of [target]'s [woundtype]."))
```
Comment (Mothblocks) [inline code/modules/surgery/healing.dm:68]: Needs to be one higher.

Multi-line procs should be written like:

```dm
proc_on_one_line(
    arg1,
    arg2,
    arg3_with_trailing_comma,
)
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by RikuTheKiller** [comment]
Uh oh surgery broke

**Pull request merged**